packages/react-chat/src/contexts/RuntimeContext/useRuntimeState.ts

```diff
@@ -72,6 +72,23 @@ export const useRuntimeState = ({ assistant, config, traceHandlers }: Settings)
 
   const reset = () => setTurns(() => []);
 
+  const getLastSystemTurn = (): TurnProps | null => {
+    const turns = getTurns();
+    for (let i = turns.length - 1; i >= 0; i--) {
+      if (turns[i].type === TurnType.SYSTEM) return turns[i];
+    }
+    return null;
+  };
+
+  const userRequestMatchingAction = (action: BaseRequest) => {
+    const systemTurn = getLastSystemTurn();
+    if (isTextRequest(action) && systemTurn?.type === TurnType.SYSTEM) {
```

Comment:
isn't this check again on `systemTurn?.type === TurnType.SYSTEM)` redundant?

Reply:
yeah, it's redundant.. i'll remove it.  
(was before i extracted the method `getLastSystemTurn` properly)